### DISCUSSION TITLE
Pointer Events in React Native

### DISCUSSION THREAD

**Discussion by lunaleaps** [react-native-community/discussions-and-proposals#557] [Ideas]
Discussion dedicated to Pointer Events as announced in this [blog post](https://reactnative.dev/blog/2022/12/13/pointer-events-in-react-native). 

If you have any questions or feedback on this work, let's discuss!

**Comment by julia-script** [comment]
This is very cool, thank you!

I'm trying to set it up and I can't make it work for the life of me..I have experience with React but not so much with react-native.
I can't get it to work on ios.

I also made this code sandbox example. I'm not sure it should work on the web anyway as the blog post doesn't directly mention it
https://codesandbox.io/s/frosty-mcclintock-kjd5vf?file=/src/index.js

any tips?

EDIT: got it to work. I stumbled in this line of code and found out I had to have Fabric enabled
https://github.com/facebook/react-native/blob/17f221c85282c2aa00cd1d0303930a76182513a1/Libraries/Inspector/DevtoolsOverlay.js#L184

maybe that's obvious for someone more familiar with react-native ecosystem, still, I think it'd be nice to mention that in the blog post..

Thank you all

**Reply by julia-script** [reply]
yea, that's what I meant

**Reply by lunaleaps** [reply]
Thanks @julia-script for the feedback! Just updated our 0.71 disclaimer to clarify that "New Architecture" refers to Fabric and linked to our docs on how to enable!

**Comment by stmiller-shamrock** [comment]
I am wondering if this will work with an Expo app or if I am just doing something wrong. I went through the process of setting the feature flags in JS, Android, and iOS. I was able to add them to a component and then no events. The other weirdness I ran into was I am not sure if the types are correct, because with TypeScript, it wouldn't allow me to add the `onPointerOver` prop but did allow pointer events not even listed in the article.

I don't know all the ins and out of Expo. I was just thinking maybe because of Expo Go, that I couldn't set those feature flags natively. I normally don't use Expo, but I am writing a tutorial on pointer events and it was easy to get started that way.

**Reply by vincentriemer** [reply]
I'm fairly certain Expo doesn't support the new architecture yet (i.e. Fabric) so even with the feature flags enabled it still would not work due to the implementation only targeting Fabric.

**Reply by stmiller-shamrock** [reply]
Cool thanks! That helps. I've been messing with this for hours.

**Comment by chadders404** [comment]
Thanks so much for all your hard work! Glad to see RN striving to keep up with the blurring of lines between desktop and mobile. The "onClick" functionality will be especially fundamental to our app. We need to trigger certain events via mouse click, that arent triggered by tapping, so I'm hoping this will be implemented soon.

**Comment by annguyenict172** [comment]
hi @lunaleaps! I noticed that support for pointer events has been marked as experimental since 2022 and is still behind a feature flag as of today. Is there any plan to release a stable version in the near future? If not, I'd love to understand what's currently blocking it from being rolled out. Thank you!
